src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores.component.ts

```diff
@@ -303,4 +323,32 @@ export class ExerciseScoresComponent implements OnInit, OnDestroy {
         // TODO: we should try to use the artemis date pipe here
         return date ? dayjs(date).format(defaultLongDateTimeFormat) : '';
     }
+
+    /**
+     * filters the displayable results based on the given range filter
+     * @param results all results for the given exercise
+     * @returns results falling into the given score range
+     */
+    filterByScoreRange(results: Result[]): Result[] {
+        if (!this.rangeFilter) {
+            return results;
+        }
+        let filterFunction;
+        if (this.rangeFilter.upperBound === 100) {
+            filterFunction = (result: Result) => !!result.score && result.score >= this.rangeFilter!.lowerBound && result.score <= this.rangeFilter!.upperBound;
+        } else {
+            filterFunction = (result: Result) => result.score !== undefined && result.score >= this.rangeFilter!.lowerBound && result.score < this.rangeFilter!.upperBound;
```

Comment:
In case of the range has an upper bound of 100%, it does not make a difference whether the score is undefined or 0, as the score would not satisfy the range [90, 100] in any case.
If the upper bound is not 100%, this does not necessarily hold, since the upper bound could be 10%. So in this case, we need to check explicitly for definition.